docs/src/3.4.MouseEvents.mdx

```diff
@@ -19,56 +20,124 @@ There are two kinds of event handlers:
+| `nonInstancedGetHitmap`    | A default `getHitmap` function for non-instanced commands.                                                                                                                   | `<Command getHitmap={nonInstancedGetHitmap} ... />`                                  |
+| `createInstancedGetHitmap` | A factory function for instanced commands. It takes the number of points per instance: for example, a triangle has 3 points, whereas a sphere has 1 point (just the center). | `<Command getHitmap={createInstancedGetHitmap({ pointCountPerInstance: 1 })} ... />` |
 
-### Hitmap for Instanced Objects
+##### Turning off interactivity for a layer
 
-Usually using instanced objects can help improve rendering performance, e.g. an instanced sphere can contain thousands of spheres. When the user clicks a single sphere, the mapped instanceId (returned as `objectId`) which was generated through `getHitmapProps` function will be returned from the event.
+All built in commands allow overriding the `getHitmap` function, and we recommend that you also allow users to override it. If a user passes in `null` or `undefined` for `getHitmap` in a command, any objects drawn by that command instance will not be visible for interaction purposes. Disabling interaction in this way may help improve performance.
 
-_Commands that currently support instanced rendering: Cubes, Points, Spheres, Triangles, Cones, Cylinders. More to be added later._
+##### Customizing the getHitmap function
+
+Each Command is passed the `reglCommand` function and `children`. Worldview calls compiled version of the `reglCommand` function with `children`. `getHitmap` takes in those `children` and returns new version of them that will be rendered into the hitmap.
+
+Some important things to remember:
+ * The first argument of `getHitmap` is the `children`.
+ * The second argument is a function that helps us assign IDs to each object.
+```js
+/*
+ * type: "instanced" if mapping multiple IDs to a single callback object, "single" if 1:1.
+ * callbackObject: the object to pass to event callbacks when this object is interacted with.
+ * count: If "instanced", how many IDs to map to the callback object.
+ * return type: an array of the IDs assigned.
+ */
+type AssignNextIdsFn = (
+  { type: "single", callbackObject: BaseShape } | { type: "instanced", callbackObject: BaseShape, count: number }
+) => Array<HitmapId>;
+```
+ * Each unique object that you want to event on should get a unique ID. If you want to map parts of a single object to multiple IDs (instancing), use the `type: "instanced"` call to `assignNextIds`.
+ * Once you have this ID, pass it to `intToRGB()` to get the color to paint your object. Don't apply any transparency or shading to your object or this will mess up the hitmap.
```

Comment:
we could pass the next color instead of the next id. Passing it into the fragment shader is an option but I had earlier opted not to do that because I thought it made the API more confusing and less explicit.

Reply:
I love @jtbandes's suggestion.  I wonder if we can have a general api for reglCommand: `hitmapColor`. For some of the complex reglCommand (e.g. MapTiles, GLTFScene, VirtualTileMaps), we need to pass down `isHitmap` anyhow to distinguish if we are drawing regular objects or hitmap objects.  This will also simplify hitmapProp generation since we can just use the original `drawProps`.

Reply:
On a second thought, we can't just use the original `drawProps` for `hitmapProps`, because we might need to do something special such as changing `scale` when drawing hitmap objects, and these kind of changes are normally specified at the consumer side.  So agree with @davidswinegar, we should not pass the color(s) to the fragment shader.